### ISSUE TITLE
Consider requiring clang-format for all non-external sources for readability.

### ISSUE THREAD

**Issue by dalecurtis** [joedrago/avif#12]
I.e., basically standardize on clang-format as the style for all non-external code. Totally up to you, so feel free to reject this "issue" with prejudice, but I feel this helps the legibility and maintainability of projects over time.

https://clang.llvm.org/docs/ClangFormat.html
https://github.com/llvm-mirror/clang/tree/master/tools/clang-format

Example implementation:
https://github.com/KhronosGroup/SPIRV-Tools/issues/1029

If I'm being selfish I'd recommend the -style=Chromium, but really having any .clang-format standardized dotfile would be beneficial :)

**Comment by joedrago** [comment]
Good idea, I've just done this now: 

https://github.com/AOMediaCodec/libavif/commit/7a9a661df3ae09dc4ed48de80d5c8827316ef060

This repo (under my personal email) is retired, now replaced with AOM's official repo here:

https://github.com/AOMediaCodec/libavif

Please put future issues over there.

**Comment by dalecurtis** [comment]
Thanks! Sorry, thought I was on that repo when posting. Did you mean to set ColumnLimit=0? That'll create infinitely long lines in some cases; see the utils.c diff in your change.

**Comment by joedrago** [comment]
Yeah, for now, unfortunately.

In our internal repos at work and in all of my personal projects, I have ancient-yet-carefully-honed uncrustify config that runs on save in my editors. I studied each clang-format config as I made this file, and tried to mimic the behavior I already had in uncrustify, but ColumnLimit aggressively changes too many lines, and if the only current fallout is one line in util.h being odd, I can live with that.

I'm not a believer in Google's 80 column hard wrap; it is way too severe in a world of modern widescreen monitors, IMO. I think a soft wrap at 100 and hard wrap at 120 feels much better, and in many cases I don't count trailing comments against the wrap restriction. clang-format is a bit more opinionated / less configurable than uncrustify, so I tried to thread the needle here, and this is where I'm at for now.

I'll see if I can come up with a reasonable ColumnLimit if I can live with, but other than that one odd line in utils.c, everything seemed to come through reasonably.
